src/pyFAI/io/integration_config.py

```diff
@@ -493,22 +516,44 @@ def from_dict(cls, dico, inplace=False):
         if not inplace:
             dico = copy.copy(dico)
         normalize(dico, inplace=True)
-        to_init = {field.name:dico.pop(field.name)
-                   for field in fields(cls)
-                   if field.name in dico}
+
+        to_init = {}
+        for field in fields(cls):
+            key = field.name
+            if key in dico:
+                value = dico.pop(key)
+                if key in cls.ENFORCED:
+                    "Enforce a specific class type"
+                    klass = field.type
+                    if value is None:
+                        to_init[key] = value
+                    elif isinstance(value, klass):
+                        to_init[key] = value
+                    elif isinstance(value, dict):
+                        to_init[key] = klass(**value)
+                    elif isinstance(value, (list, tuple)):
+                        to_init[key] = klass(*value)
```

Comment:
```suggestion
                    elif isinstance(value, (dict, list, tuple)):
                        to_init[key] = klass(*value)
```

Reply:
Does this work ? in the case `value` is a dict: 
`*value` is the same as `*list(value.keys())` which is not what is expected ... or did I miss something ? Moreover, I don't think the order of the keys is guaranteed.

Reply:
No, my bad, I didn't read carefully.